### PR TITLE
feat: update benchmark

### PR DESCRIPTION
**Pull request by TonyRL**
## PR 的功能

- Add results from `@napi-rs/pinyin` for comparison.
- Update the accuracy benchmark script to use the program output from `pinyin` and `@napi-rs/pinyin` instead of hard-coded output following the instructions in both [pinyin](https://github.com/hotoo/pinyin) and [@napi-rs/pinyin](https://github.com/Brooooooklyn/pinyin), which installs the latest version as specified.
- Update the speed benchmark script to include benchmarks for `pinyin-pro`, `pinyin`, and `@napi-rs/pinyin`.
- Update the benchmark results.

## 你的预期是什么

To update the benchmark results according to the instructions of each package. As of proposing this PR, it's `3.14.0` for `pinyin-pro`, `3.0.0-alpha.5` for `pinyin`, and `1.7.1` for `@napi-rs/pinyin`.

## 是否进行了详细的自测？

Yes.

### Test Environment
Gitpod Standard Workspace ([up to 4 cores, up to 8GB RAM](https://www.gitpod.io/docs/configure/workspaces/workspace-classes))
Node.js v18.16.0.

### Test results
<details><summary>Accuracy</summary>

accuracy
</details>

<details><summary>5000 words</summary>

5k
</details>

<details><summary>10000 words</summary>

10k
</details>

<details><summary>1 million words</summary>

1m
`pinyin` cannot finish the benchmark within a reasonable time (1 minute).
</details>

The updated results are calculated by taking the average of 5 trials. Your mileage may vary.

### CODE REVIEW

**Comment by zh-lx** [comment]
感谢 PR，关于性能测试方面，`pinyin-pro` 是默认开启了多音字分词能力的，所以 `pinyin` 和 `@napi-rs/pinyin` 的性能测试需要开启 `segment: 'true'` 选项比较才有意义，可以修改一下

**Comment by zh-lx** [comment]
> > 感谢 PR，关于性能测试方面，`pinyin-pro` 是默认开启了多音字分词能力的，所以 `pinyin` 和 `@napi-rs/pinyin` 的性能测试需要开启 `segment: true` 选项比较才有意义，可以修改一下
> 
> I think I did 🤔.
> 
> https://github.com/zh-lx/pinyin-pro/blob/55b82291bca0424530e92493dd111983fb8480b0/benchmark/accuracy.js#L10
> 
> 
> https://github.com/zh-lx/pinyin-pro/blob/55b82291bca0424530e92493dd111983fb8480b0/benchmark/accuracy.js#L20

准确率测试 accuracy 中你开启了，但是性能测试 speed.js 你没有对 `pinyin` 和 `@napi-rs/pinyin` 开启分词，会使这两个包的性能看上去虚高太多。

File: README.md [hunk 1]
```diff
@@ -68,9 +68,9 @@ pinyin('汉语拼音', { pattern: 'final' }); // 'àn ǔ īn īn'
 | 对比项              | pinyin           | pinyin-pro | @napi-rs/pinyin |
 | ------------------ | ---------------- | ---------- | --------------- |
 | 准确率              | 97.844%          | 99.744%    | 97.433%         |
-| 5k 字转换所需时间    | 4.879ms          | 6.648ms    | 12.386ms        |
-| 1w 字转换所需时间    | 11.240ms         | 18.315ms   | 27.468ms        |
-| 100w 字转换所需时间  | 392.283ms        | 624.481ms  | 3.782s          |
+| 5k 字转换所需时间    | 489.252ms        | 5.909ms    | 115.723ms       |
+| 1w 字转换所需时间    | 511.573ms        | 15.260ms   | 115.887ms       |
+| 100w 字转换所需时间  | 内存溢出转换失败    | 595.131ms  | 570.994s        |
```
Comment (TonyRL) [inline README.md:73]: Updated the benchmark results with the option `segment: true`.
`pinyin` failed to complete it. Interestingly, `@napi-rs/pinyin` runs faster with `segment: true` than its default.

Comment (TonyRL) [inline README.md]: Updated the benchmark results with the option `segment: true`.
`pinyin` failed to complete it. Interestingly, `@napi-rs/pinyin` runs faster with `segment: true` than its default.

The screenshots of the test result are also updated accordingly.

**Review by zh-lx — APPROVED** [review]

**Pull request merged**